### PR TITLE
Fixing CircleCI image generation

### PR DESCRIPTION
**Pull request by CatGarab**
- Updating the CircleCI build script (`build.sh`) to use our BBB target instead of the old legacy one. 
- Copying a compressed version of the final SD card image to the artifacts folder instead of the original, since it's waaaaaay smaller.
- Updating the BBB's post-build logic to solely use `genimage` to create the SD card image, rather than a loopback device (which isn't supported by CircleCI or our Docker image)
- Updating the BBB's config to include the new needed tools during the build process

### CODE REVIEW

**Comment by CatGarab** [comment]
[CircleCI building Beaglebone sd card image](https://trello.com/c/Sn9uK8AJ/2-circleci-building-beaglebone-sd-card-image)

File: build.sh [hunk 1]
```diff
@@ -11,6 +11,8 @@ cd .. #cd out of the kubos-linux-build directory
 
 kubos update
 
+apt-get install dosfstools
```
Comment (plauche) [inline build.sh]: We probably want to put this `apt-get install` into the docker provisioning script

File: post.sh [hunk 2]
```diff
@@ -1,4 +1,4 @@
 #!/bin/bash
 
-artifact_url="https://${CIRCLE_BUILD_NUM}-73321966-gh.circle-artifacts.com/0${CIRCLE_ARTIFACTS}/sdcard.img"
+artifact_url="https://${CIRCLE_BUILD_NUM}-73321966-gh.circle-artifacts.com/0${CIRCLE_ARTIFACTS}/kubos-linux.tar.gz"
```
Comment (plauche) [inline post.sh]: @kyleparrott what is the `73321966-gh` part of this url?

Comment (kyleparrott) [reply post.sh]: Best guess is it's the Circle Project ID (from the gh integration), or some form of primary key for our project. Whatever it truly is, it's constant between builds. 

Example from the full_build.log artifact:

build #272: https://272-73321966-gh.circle-artifacts.com/0/tmp/circle-artifacts.EtFmO6y/full_build.log
build #247: https://247-73321966-gh.circle-artifacts.com/0/tmp/circle-artifacts.iUtr186/full_build.log

Comment (kyleparrott) [reply post.sh]: Also the "0" on `../0${CIRCLE_ARTIFACTS}/..` is the container number from the build, if we start to integrate more containers..

Comment (plauche) [reply post.sh]: :+1:

**Pull request merged**